### ISSUE TITLE
WP 5.6.1 FontSize Select doesn't Work

### ISSUE THREAD

**Issue by voregnev** [WordPress/gutenberg#28738]
## Description
Select control Font Size for Gutenberg typography doesn't work.
## Step-by-step reproduction instructions
Go to Console -> Add New Post
add new Paragraph
Type some text
Try to change FontSize by pseudo-select control
Change font by direct font size number field

## Expected behaviour
Select FontSize and Number FontSize controls should be syncronized

## Actual behaviour
Select FontSize at any option reset FontSize
## Screenshots or screen recording (optional)

![fontsize](https://user-images.githubusercontent.com/37963761/106907861-c27bdd80-670f-11eb-91ff-decfa4683577.gif)

## WordPress information
- WordPress version: 5.6.1
- Gutenberg version: unknown
- Are all plugins except Gutenberg deactivated? No
- Are you using a default theme (e.g. Twenty Twenty-One)? No

## Device information
- Device: Desktop
- Operating system: "Windows 7" MAc Yosemite
- Browser:  "Chrome 86.0" or Safari

**Comment by DietteJanssen** [comment]
i am also experiencing this SAME issue when updating to the latest wordpress version.

**Comment by davecaputo** [comment]
Ditto, same issue immediately after update.

**Comment by gwwar** [comment]
This is also reported in https://core.trac.wordpress.org/ticket/52449 and appears to be a regression from https://github.com/WordPress/gutenberg/pull/26705#issuecomment-773068915

**Comment by MiltonScholarSupreme** [comment]
I thought I had accidentally broken my site! Haha. I'm new to Wordpress; how long does it take for these issue to be resolved? After updating I'm having the same issue. However, I'm still able to change the font size with the by entering the size next to the selector.

**Comment by Jamiewarb** [comment]
We're also having the same problem across all our block editor sites that updated to 5.6.1

**Comment by talldan** [comment]
@voregnev @MiltonScholarSupreme @Jamiewarb This bug only happens in some themes. What's a good theme to reproduce this issue with?

**Comment by anarieldesign** [comment]
Same here, I tested it with many themes and it has the same bug. It was working before this update.

**Comment by overclokk** [comment]
I'm having the same issue, also tested with default theme 2021 and no plugin actived.

**Comment by talldan** [comment]
@anarieldesign Which themes are affected? We need to know in order to fix the issue.

**Comment by ppezier** [comment]
> 
> @anarieldesign Which themes are affected? We need to know in order to fix the issue.

I don't think it's a theme-related issue.
I've experienced the exact same bug with both twenty-one and a home-made theme (neither add_theme_support( 'disable-custom-font-sizes' ) nor add_theme_support( 'editor-font-sizes', ... ) are used in this theme).

With WP 5.6, the behaviour is normal and since 5.6.1, all entries in "Font size" are visually the same + choosing one doesn't affect the content of the paragraph block.

**Comment by voregnev** [comment]
Now I install clear WP 5.6.1 with Default 21 theme
Do you need access?

**Comment by darwin370** [comment]
Same here on all 5.6.1 sites, rollback to 5.6 fixes the issue.
> This bug only happens in some themes. What's a good theme to reproduce this issue with?
@talldan Twenty Nineteen, Twenty Twenty, Twenty Twenty-One, use Step-by-step reproduction instructions in issue description.

**Comment by talldan** [comment]
Right, I see now why I'm confused. Sorry for that.

> choosing one doesn't affect the content of the paragraph block.

This seems to have been fixed in the Gutenberg plugin. We'll need to track down where it was fixed.

> all entries in "Font size" are visually the same

This part is still a separate bug that only happens in some themes, which is what https://github.com/WordPress/gutenberg/pull/28761 addresses.

**Comment by CHEWX** [comment]
Anyone know a good patch for this bug other than rolling back to 5.6 or installing the Gutenberg plugin?

**Comment by gwwar** [comment]
Confirmed that this is broken in 5.6.1, but fixed with Gutenberg plugin enabled. I set a breakpoint, and in 5.6.1 we're passing in a NaN value to the fontsize onChange function.

<img width="635" alt="Screen Shot 2021-02-11 at 10 01 04 AM" src="https://user-images.githubusercontent.com/1270189/107679475-75d55c80-6c51-11eb-9008-a9a96d3e02d4.png">

This was fixed in
<img width="1521" alt="Screen Shot 2021-02-11 at 10 12 11 AM" src="https://user-images.githubusercontent.com/1270189/107679607-a0271a00-6c51-11eb-9469-4bd287bf8ed5.png">
https://github.com/WordPress/gutenberg/pull/26475/files#diff-3b85bc4bfdaea5cd9cb6173bf9f52656ce4f656e5f846442fd15185a5b78fb1dR102-R106 

Notice how we check for units, before forcing a number conversion.

@talldan I'd recommend cherrypicking https://github.com/WordPress/gutenberg/pull/26475 for a 5.6.2 fix.
